packages/metals-vscode/src/debugger/scalaDebugger.ts

```diff
@@ -191,13 +199,18 @@ class ScalaDebugServerFactory implements vscode.DebugAdapterDescriptorFactory {
         session.configuration
       );
 
-      if (debugSession === undefined) {
-        return null;
-      } else {
+      if (debugSession !== undefined && debugSession.uri !== undefined) {
         return debugServerFromUri(debugSession.uri);
+      } else if (
+        debugSession !== undefined &&
+        debugSession.error !== undefined
+      ) {
+        throw new Error(debugSession.error);
+      } else {
+        throw new Error("Could not start debug adapter");
       }
```

Comment:
`else { throw new Error("Could not start debug adapter") }` seems redundant, as there's fallback return at the end already